src/cmdc/datasets/official/CA/counties/imperial.py

```diff
@@ -0,0 +1,90 @@
+import requests
+import pandas as pd
+
+from cmdc.datasets.official.base import ArcGIS
+
+
+class Imperial(ArcGIS):
+    """
+    Imperial publishes their county level data in a dashboard that can
+    be found at:
+
+    http://www.icphd.org/health-information-and-resources/healthy-facts/covid-19/
+
+    They don't seem to provide any simple interface to their data...
+    However, one can directly request data from arcgis itself which is
+    the route we take here. They seem to change the service that they
+    use to store their data... May need to think about how to deal with
+    that
+
+    The base arcgis list can be found at:
+
+    https://services7.arcgis.com/RomaVqqozKczDNgd/ArcGIS/rest/services
+    """
+    ARCGIS_ID = "RomaVqqozKczDNgd"
+    FIPS = 6073
+
+    def __init__(self):
+        self.hospitaloutfields = {
+            "DATE": "dt",
+            "LICENSED_BEDS": "hospital_beds_capacity_count",
+            "HOSPITALIZED_COVID_CONFIRMED_PA": "hospital_beds_in_use_covid_confirmed",
+            "HOSPITALIZED_SUSPECTED_COVID_PA": "hospital_beds_in_use_covid_suspected",
+            "ICU_BEDS": "icu_beds_capacity_count",
+            "ICU_OTHER_PATIENTS": "icu_beds_in_use_other",
+            "ICU_SUSPECTED_COVID_PATIENTS": "icu_beds_in_use_covid_suspected",
+            "ICU_COVID_CONFIRMED_PATIENTS": "icu_beds_in_use_covid_confirmed",
+            "MECHANICAL_VENTILATORS": "ventilators_capacity_count",
+            "MECHANICAL_VENTILATORS_IN_USE": "ventilators_in_use_any",
+            "MECHANICAL_VENTILATORS_IN_USE_1": "ventilators_in_use_covid_suspected",
+            "MECHANICAL_VENTILATORS_IN_USE_F": "ventilators_in_use_covid_confirmed",
+        }
+
+        # Default parameter values
+        params = {
+            "where": "0=0",
+            "returnGeometry": "false",
+            "f": "pjson"
+        }
+
+        super(Imperial, self).__init__(params=params)
+
+        return None
+
+    def get_hospital_data(self):
+
+        self.params.update({"outFields": ",".join(self.hospitaloutfields)})
+        _url = self.arcgis_query_url("EKlhM", 0, 7)
+        req = requests.get(_url, params=self.params)
+
+        df = pd.DataFrame.from_records(
+            [x["attributes"] for x in req.json()["features"]]
+        ).rename(columns=self.hospitaloutfields)
+        df["vintage"] = pd.datetime.today()
+        df["dt"] = df["dt"].map(
+            lambda x: pd.datetime.fromtimestamp(x/1000)
+        )
+        df["fips"] = self.FIPS
+
+        df["icu_beds_in_use_covid_total"] = df.eval(
```

Comment:
Again... comment about the overwriting with NaN issue.

Also, we seem to repeat this code often. Does it make sense to put it in a method in the base classes and call it from there?

Reply:
It could be a little different in each of the county cases because they're reporting slightly different things:

Notice here that they've reported the total number of ICU beds in use by non-covid patients and I have to find the total number of ICU beds by combining different columns than other case.